docs/macros.py

```diff
@@ -0,0 +1,77 @@
+"""Macros and filters made available in Markdown pages."""
+
+import asyncio
+from itertools import chain
+from pathlib import Path
+
+import httpx
+import toml
+from jinja2 import StrictUndefined
+from jinja2.sandbox import SandboxedEnvironment
+from pip._internal.commands.show import search_packages_info  # noqa: WPS436 (no other way?)
+
+
+def get_credits_data() -> dict:
+    """
+    Return data used to generate the credits file.
+
+    Returns:
+        Data required to render the credits template.
+    """
+    project_dir = Path(__file__).parent.parent
+    metadata = toml.load(project_dir / "pyproject.toml")["tool"]["poetry"]
+    lock_data = toml.load(project_dir / "poetry.lock")
+    project_name = metadata["name"]
+
+    poetry_dependencies = chain(metadata["dependencies"].keys(), metadata["dev-dependencies"].keys())
+    direct_dependencies = {dep.lower() for dep in poetry_dependencies}
+    direct_dependencies.remove("python")
+    indirect_dependencies = {pkg["name"].lower() for pkg in lock_data["package"]}
+    indirect_dependencies -= direct_dependencies
+    dependencies = direct_dependencies | indirect_dependencies
+
+    packages = {}
+    for pkg in search_packages_info(dependencies):
+        pkg = {_: pkg[_] for _ in ("name", "home-page")}
+        packages[pkg["name"].lower()] = pkg
+
+    loop = asyncio.get_event_loop()
+    client = httpx.AsyncClient()
+    to_get = [dep for dep in dependencies if dep not in packages]
+    coroutines = [client.get(f"https://pypi.python.org/pypi/{dep}/json") for dep in to_get]
+    responses = loop.run_until_complete(asyncio.gather(*coroutines))
+    loop.run_until_complete(client.aclose())
```

Comment:
I'm getting `RuntimeError: This event loop is already running` when this is hit for the 2nd time within `mkdocs serve`

Reply:
Ha! I missed that. Will run some tests and fix it.

Reply:
Damn I struggle to find a solution to this.

Reply:
An idea. Maybe drop the download part altogether? There are only like 5 packages that are queried. Either just deal without that information or hardcode it for those packages.

Reply:
You know what you're right. The packages that pip could not find are the one that were not installed because they're now part of the standard lib or only useful or Windows. Let's drop them. I guess my ideas just need to be confronted from time to time :grinning:

Reply:
Don't forget to drop the httpx dependency at the end 🤭